### ISSUE TITLE
[HOLD for payment 2024-08-01] [$250] Emoji - Thumb emoji is displayed in frequently used every time when change the color

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#45262]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.6-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** gocemate+a614@gmail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Go to any chat
2. Type :+1 emoji and send
3. Click on emoji symbol> Change default skin tone
4. Type :+1 emoji and send
5. Click on emoji symbol
6. Check Frequently used section

## Expected Result:
Only one thumb emoji should be displayed

## Actual Result:
Thumb emoji is displayed in frequently used every time when change the color and send to chat

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/a06cf3c5-52a0-4b3c-a1c1-4568dfb813ce

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019afad155d77a1825</li>
        <li>Upwork Job ID: 1812957552878135707</li>
        <li>Last Price Increase: 2024-07-15</li>
<li>Automatic offers: </li>
<ul>
<li>nyomanjyotisa | Contributor | 103163861</li>
<li>situchan | Reviewer | 103164237</li>
<li>c3024 | Contributor | 103164238</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @johncschuster</details>

**Comment by izarutskaya** [comment]
We think this issue might be related to the #vip-vsb.

**Comment by c3024** [comment]
Since the response from backend has duplicates this should ideally be fixed from backend. 

However, I found another issue here. We are not adding the recently used emojis to the frequently used emojis optimistically. This does not fall directly under the scope of the bug. But we might consider fixing that. This would be a frontend fix.

🎀 👀 🎀 C+ Reviewed for Internal Engineer's decision

**Comment by Beamanator** [comment]
OK it looooks like this could have been caused by some recent changes in https://github.com/Expensify/App/pull/42208? (cc @jasperhuangg ) mind taking a look?

**Comment by jasperhuangg** [comment]
Oh woops, this definitely looks like a regression from my `UpdateFrequentlyUsedEmojis` performance improvement PRs. I'll look into this. Definitely a back-end bug.

**Comment by jasperhuangg** [comment]
@c3024 unassigning you since this requires a back-end fix

**Comment by jasperhuangg** [comment]
Looking into this today

**Comment by jasperhuangg** [comment]
Hmm I investigated a fix in the back-end, but I don't think it's worth going down that rabbit hole since C++ isn't set up to handle emoji codes well.

I looked over the proposals and think the one from @nyomanjyotisa makes the most sense. Going to go ahead and assign them. Let's make sure to leave some good comments in your PR explaining why we're doing so. Thanks everyone!

**Comment by c3024** [comment]
@jasperhuangg I was the original c+ for this issue. Could you please assign me back?

**Comment by c3024** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [x]  [@c3024] The PR that introduced the bug has been identified. Link to the PR: No specific PR can be held responsible. This stemmed likely from a [backend PR and the author of the PR is aware of it.](https://github.com/Expensify/App/issues/45262#issuecomment-2231536937)
> * [x]  [@c3024] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: Same as above
> * [x]  [@c3024] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: No discussion was started because this could not be identified earlier.
> * [x]  [@c3024] Determine if we should create a regression test for this bug. Yes
> * [x]  [@c3024] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

### Regression Test Proposal
1. Go to any chat
2. Type `:+1:` emoji (or any other emoji) in the composer and send
3. Click on emoji picker in the composer -> Change default skin tone to a different skin tone > Close the emoji picker
4. Type the same emoji in Step 2 in the composer and send
5. Click on emoji picker in the composer again
6. Verify there is no duplicate of any emoji in `Frequently Used` section
